src/test/java/org/gridsuite/ds/server/service/client/dynamicmapping/DynamicMappingClientTest.java

```diff
@@ -0,0 +1,157 @@
+public class DynamicMappingClientTest extends AbstractRestClientTest {
+    // mapping names
+    public static final String MAPPING_NAME_01 = "_01";
+
+    // directories
+    public static final String DATA_IEEE14_BASE_DIR = "/data/ieee14";
+    public static final String INPUT = "input";
+    public static final String MODELS_GROOVY = "models.groovy";
+    private static final int DYNAMIC_MAPPING_PORT = 5036;
+
+    private DynamicMappingClient dynamicMappingClient;
+
+    @Override
+    @NotNull
+    protected Dispatcher getDispatcher() {
+        return new Dispatcher() {
+            @NotNull
+            @Override
+            public MockResponse dispatch(@NotNull RecordedRequest recordedRequest) {
+                String path = Objects.requireNonNull(recordedRequest.getPath());
+                String baseUrl = DELIMITER + API_VERSION + DELIMITER + DynamicMappingClient.DYNAMIC_MAPPING_SCRIPT_CREATE_END_POINT + DELIMITER;
+                baseUrl = baseUrl.replace("//", "/");
+                String method = recordedRequest.getMethod();
+                MockResponse response = new MockResponse().setResponseCode(HttpStatus.NOT_FOUND.value());
+                // scripts/from/{mappingName}
+                if ("GET".equals(method)
+                        && path.matches(baseUrl + ".*")) {
+                    // take {mappingName} at the last
+                    String mappingName = emptyIfNull(recordedRequest.getRequestUrl().pathSegments()).stream().reduce((first, second) -> second).orElse("");
+                    if (MAPPING_NAME_01.equals(mappingName)) {
+                        String scriptJson;
+                        try {
+                            // load models.groovy
+                            String scriptPath = Paths.get(DATA_IEEE14_BASE_DIR, mappingName, INPUT, MODELS_GROOVY).toString();
+                            InputStream scriptIS = getClass().getResourceAsStream(scriptPath);
+                            byte[] scriptBytes;
+                            scriptBytes = StreamUtils.copyToByteArray(scriptIS);
+                            String script = new String(scriptBytes, StandardCharsets.UTF_8);
+
+                            // load models.par
+                            String parametersFilePath = Paths.get(DATA_IEEE14_BASE_DIR, mappingName, INPUT, MODELS_PAR).toString();
+                            InputStream parametersFileIS = getClass().getResourceAsStream(parametersFilePath);
+                            byte[] parametersFileBytes;
+                            parametersFileBytes = StreamUtils.copyToByteArray(parametersFileIS);
+                            String parametersFile = new String(parametersFileBytes, StandardCharsets.UTF_8);
+
+                            Script scriptObj =  new Script(
+                                    mappingName + "-script",
+                                    mappingName,
+                                    script,
+                                    new Date(),
```

Comment:
Maybe it would be better/more predictable to have a fixed date.

Reply:
Done! I use "01/01/2023" as a fixed date for testing